waiter/src/waiter/client_tools.clj

```diff
@@ -191,17 +190,27 @@
             (log/warn (RuntimeException.))))
         (assoc request-headers "x-cid" new-cid)))))
 
-(defn- add-cookies
-  [waiter-url cookies ^CookieStore cs]
-  {:pre [(not (str/blank? waiter-url))]}
-  (doseq [cookie cookies]
-    (cookies/add-cookie! cs (str HTTP-SCHEME waiter-url) cookie)))
-
 (defn strip-trailing-slash
   "If s ends with /, returns s with the / stripped"
   [s]
   (cond-> s (str/ends-with? s "/") (subs 0 (- (count s) 1))))
 
+(defn parse-set-cookie-string
+  "Parse the value of a single Set-Cookie header."
+  [string-value]
+  (when string-value
+    (let [cookie-list (java.net.HttpCookie/parse string-value)]
+      (map (fn [^java.net.HttpCookie cookie]
+             {:name (.getName cookie)
+              :value (.getValue cookie)}) cookie-list))))
+
+(defn parse-cookies
```

Comment:
Should parse-cookies handle exception while parsing the set-cookie value?

Reply:
I think we should have unit tests for `parse-cookies`.

Reply:
I don't think we need to unit tests in client tools, since it's just used in tests anyway. But if @sradack wants to add them I'm not opposed

Reply:
What exception did you have in mind that should be handled locally?

Reply:
Added unit tests.

Reply:
My bad, I didn't realize this was only used in the tests and not while handling requests.